src/StreamRequest.jl

```diff
@@ -48,12 +48,20 @@ function request(::Type{StreamLayer}, io::IO, request::Request, body;
     end
 
     aborted = false
+    write_error = nothing
     try
 
         @sync begin
 
             if iofunction == nothing
-                @async writebody(http, request, body)
+                @async try
+                    writebody(http, request, body)
+                catch e
+                    write_error = e
+                    @warn("Error in @async writebody task",
+                          exception=(write_error, catch_backtrace()))
```

Comment:
I think this should be `stacktrace(catch_backtrace())`

Reply:
I did it this way because the `@warn` macro seems to do special processing of `Tuple{Exception,BackTrace}`. i.e. it prints out a nice stack-trace.